template-overlay.nix

```
@@ -0,0 +1,51 @@
+## File initially generated by createOverlay
+## and then supposedly modified manually.
+## Some hints for manual modifications are in the file,
+## but the full doc is on nixos / nix packages website:
+## https://nixos.org/manual/nixpkgs/stable/#sec-language-coq
+
+{ lib, mkCoqDerivation, which, coq
+  ## declare extra dependencies here, to be used in propagateBuildInputs e.g.
+  # , mathcomp, coq-elpi
+  , version ? null, shim ? false }:
+
+with lib; mkCoqDerivation {
+  pname = "template";
+  ## you can configure the domain, owner and repository, the default are:
+  # repo = "template";
+  # owner = "coq-community";
+  # domain = "github.com";
+
+  inherit version;
+  defaultVersion = with versions; switch coq.coq-version [
+    ## Example of possible dependencies
+    # { case = range "8.13" "8.14"; out = "1.2.0"; }
+    ## other predicates are `isLe v`, `isLt v`, `isGe v`, `isGt v`, `isEq v` etc
+  ] null;
+
+  ## Declare existing releases
+  ## leave sha256 empty at first and then copy paste
+  ## the resulting sha given by the error message
+  # release."1.1.1".sha256 = "";
+  ## if the tag is not exactly the version number you can amend here
+  # release."1.1.1".rev = "1.1.1"; # e.g. `= "v1.1.1"`
```

I would simplify down to:
```suggestion
  ## if the tag is not exactly the version number you can amend like this
  # release."1.1.1".rev = "v1.1.1";
```

---

I want to preserve the invariant that commented text is the default (most of the time)... but maybe it's silly in this case... Or I can mention it explicitly in the text rather than making it a single `#` commented line.

---

OK, I understand, but I think it would be more useful to not maintain this invariant and instead prioritize making the commented text useful.

---

You are right!